app/lib/actions/katello/repository/remove_content.rb

```diff
@@ -33,25 +35,33 @@ def plan(repository, content_units, options = {})
             plan_action(pulp_action, :pulp_id => repository.pulp_id,
                                      :clauses => {:association => {'unit_id' => {'$in' => uuids}}
             })
-
-            plan_self(:repository_id => repository.id, :user_id => ::User.current.id,
-                      sync_capsule: sync_capsule)
+            plan_self
             plan_action(CapsuleGenerateAndSync, repository) if sync_capsule
           end
         end
 
+        def create_sub_plans
+          trigger(Actions::Katello::Repository::MetadataGenerate,
+                  ::Katello::Repository.find(input[:repository_id]))
+        end
+
+        def trigger(*args)
```

Comment:
This method gets called more than just from line 44. It will fail during those 'extra' calls because the args are empty. There's no check for this [in dynflow](https://github.com/Dynflow/dynflow/blob/master/lib/dynflow/action/with_sub_plans.rb#L72), and the method is public, so I assume overriding it is intended, though it also may just be a bug in dynflow.